client/src/utils.js

```diff
@@ -1,26 +1,41 @@
 import { server_url } from './config.json'
 
-export async function request(path="", method="GET", data=null){
+
+export const log_event = new CustomEvent('user-log');
+
+export async function request(path="", method="GET", data=null) {
     let myHeaders = new Headers();
     myHeaders.append('Content-Type', 'application/json');
     if(sessionStorage['Authorization-token']){
         myHeaders.append('Authorization', 'Token ' + sessionStorage['Authorization-token']);
-    }
+    };
 
-    let response = await fetch(server_url + path, {
+    const response = await fetch(server_url + path, {
       method: method,
       headers: myHeaders,
       mode: 'cors',
       body: data
     });
-    let result = await response.json();
-    return result
+    const result = await response.json();
+    return result;
 }
 
-export function emailValidation(email){
-    return /.+@{1}.+/.test(email);
-}
+export const emailValidation = email => /.+@{1}.+/.test(email);
 
-export function isLogged() {
-  return typeof (sessionStorage['Authorization-token']) !== 'undefined'
-};
+export const isLogged = () => (typeof(sessionStorage['Authorization-token']) !== 'undefined');
+
+export function dateFromString(date_str) {
```

Comment:
[moment.js](https://momentjs.com/) is very good JS library to work with datetimes, and it can convert dates from [strings](https://momentjs.com/docs/#/parsing/string/). You may try.